deploy/rbac-permissions-operator-config/05-osd-readers.ClusterRole.yaml

```diff
@@ -87,3 +66,281 @@ rules:
   verbs:
   - get
   - list
+### END
+### Start - Allow read permissions to more resources
+# https://issues.redhat.com/browse/OSD-5537
+- apiGroups:
+  - admissionregistration.k8s.io
+  resources:
+  - '*'
+  verbs:
+  - get
+  - list
+  - watch
+- apiGroups:
+  - apiextensions.k8s.io
+  resources:
+  - customresourcedefinitions
+  verbs:
+  - get
+  - list
+  - watch
+- apiGroups:
+  - apiregistration.k8s.io
+  resources:
+  - apiservices
+  verbs:
+  - get
+  - list
+  - watch
+- apiGroups:
+  - authentication.k8s.io
+  resources:
+  - tokenreviews
+  verbs:
+  - get
+  - list
+  - watch
+- apiGroups:
+  - authorization.k8s.io
+  resources:
+  - selfsubjectaccessreviews
+  - subjectaccessreviews
+  - selfsubjectrulesreviews
+  verbs:
+  - get
+  - list
+  - watch
+- apiGroups:
+  - authorization.openshift.io
+  resources:
+  - clusterrolebindings
+  - subjectaccessreviews
+  - resourceaccessreviews
+  - clusterroles
+  verbs:
+  - get
+  - list
+  - watch
+- apiGroups:
+  - autoscaling.openshift.io
+  resources:
+  - clusterautoscalers
+  verbs:
+  - get
+  - list
+  - watch
+# osd-readers can view oauths, clusterversions, infrastructures, apiservers, consoles,
+#    authentications, builds, clusteroperators, dnses, schedulers, projects,
+#    featuregates, images, ingresses, networks, operatorhubs, proxies
+#    https://issues.redhat.com/browse/OSD-4298
+- apiGroups:
+  - config.openshift.io
+  resources:
+  - '*'
+  verbs:
+  - get
+  - list
+  - watch
+# osd-readers can view consolelinks, consolenotifications, consoleexternalloglinks,
+#    consoleyamlsamples
+- apiGroups:
+  - console.openshift.io
+  resources:
+  - '*'
+  verbs:
+  - get
+  - list
+  - watch
+# osd-readers can view flowschemas and prioritylevelconfigurations
+- apiGroups:
+  - flowcontrol.apiserver.k8s.io
+  resources:
+  - '*'
+  verbs:
+  - get
+  - list
+  - watch
+- apiGroups:
+  - image.openshift.io
+  resources:
+  - images
+  - imagesignatures
+  verbs:
+  - get
+  - list
+  - watch
+# osd-readers can view configs and imagepruners
+- apiGroups:
+  - imageregistry.operator.openshift.io
+  resources:
+  - '*'
+  verbs:
+  - get
+  - list
+  - watch
+# osd-readers can view machineconfigs, machineconfigpools, kubeletconfigs,
+#    controllerconfigs
+- apiGroups:
+  - machineconfiguration.openshift.io
+  resources:
+  - '*'
+  verbs:
+  - get
+  - list
+  - watch
```

Comment:
We should remove this.  Machineconfig will include anything we want to land on the hosts, including any secrets.  I hesitate to grant this without a use case driving it.

Reply:
I dont have a use case in mind for 'osd-reader' but I think it is essential for SREP to have at least read access to machineconfig and machineconfigpools to debug alerts relating to a node going missing/unreachable in the cluster.

In that case do you it would be ok to remove it from here and add read access to machineconfigs and machineconfigpool to srep-admins-cluster role for backplane?

Reply:
Yea, adding that to srep specifically makes sense.

Reply:
done